### PR TITLE
Migrate Avatar to new Compound component

### PR DESCRIPTION
**Pull request by germain-gg**
Uses https://compound.element.io/?path=/story/compound-web_avatar--round

Screenshot 2023-08-09 at 15 43 51
Screenshot 2023-08-09 at 15 44 05

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Migrate Avatar to new Compound component ([\#11393](https://github.com/matrix-org/matrix-react-sdk/pull/11393)).

### CODE REVIEW

File: src/components/views/avatars/BaseAvatar.tsx [hunk 1]
```diff
@@ -109,119 +112,32 @@ const BaseAvatar: React.FC<IProps> = (props) => {
...
         width = 40,
-        height = 40,
         resizeMethod = "crop", // eslint-disable-line @typescript-eslint/no-unused-vars
-        defaultToInitialLetter = true,
         onClick,
         inputRef,
         className,
+        type = "round",
         altText = _t("Avatar"),
-        ariaLabel = _t("Avatar"),
```
Comment (t3chguy) [inline src/components/views/avatars/BaseAvatar.tsx:118]: Losing this seems bad, this is the aria-label for the button which wraps clickable avatars

**Review by t3chguy — CHANGES_REQUESTED** [review]

**Comment by t3chguy** [comment]
Any chance of screenshots of more variants, with globe cutout, with online indicator cutout, user info (large), space home, etc etc

File: src/components/views/avatars/BaseAvatar.tsx [hunk 2]
```diff
@@ -108,119 +111,32 @@ const BaseAvatar: React.FC<IProps> = (props) => {
...
-                aria-hidden="true"
-                data-testid="avatar-img"
-            />
-        );
-
-        if (onClick) {
-            return (
-                <AccessibleButton
-                    aria-label={ariaLabel}
-                    aria-live="off"
```
Comment (t3chguy) [inline src/components/views/avatars/BaseAvatar.tsx:160]: Losing these seems bad

File: src/components/views/avatars/BaseAvatar.tsx [hunk 3]
```diff
@@ -108,119 +111,32 @@ const BaseAvatar: React.FC<IProps> = (props) => {
...
-                    width: toPx(width),
-                    lineHeight: toPx(height),
-                }}
-            >
-                {initialLetter}
-            </span>
-        );
-        const imgNode = (
-            <img
-                loading="lazy"
```
Comment (t3chguy) [inline src/components/views/avatars/BaseAvatar.tsx:141]: Losing this seems bad

File: src/components/views/avatars/BaseAvatar.tsx [hunk 4]
```diff
@@ -108,119 +111,32 @@ const BaseAvatar: React.FC<IProps> = (props) => {
...
-                src={AvatarLogic.defaultAvatarUrlForString(idName || name)}
-                alt=""
-                title={title}
-                onError={onError}
-                style={{
-                    width: toPx(width),
-                    height: toPx(height),
-                }}
-                aria-hidden="true"
-                data-testid="avatar-img"
```
Comment (t3chguy) [inline src/components/views/avatars/BaseAvatar.tsx:152]: Losing these seems bad

File: src/components/views/avatars/BaseAvatar.tsx [hunk 5]
```diff
@@ -108,119 +111,32 @@ const BaseAvatar: React.FC<IProps> = (props) => {
...
-                    {imgNode}
-                </AccessibleButton>
-            );
-        } else {
-            return (
-                <span
-                    className={classNames("mx_BaseAvatar", className)}
-                    ref={inputRef}
-                    {...otherProps}
-                    role="presentation"
```
Comment (t3chguy) [inline src/components/views/avatars/BaseAvatar.tsx:177]: Losing this seems bad

File: src/components/views/avatars/BaseAvatar.tsx [hunk 6]
```diff
@@ -20,33 +20,36 @@ limitations under the License.
...
+    idName?: React.ComponentProps<typeof Avatar>["id"]; // ID for generating hash colours
     title?: string; // onHover title text
     url?: string | null; // highest priority of them all, shortcut to set in urls[0]
     urls?: string[]; // [highest_priority, ... , lowest_priority]
+    type?: React.ComponentProps<typeof Avatar>["type"];
     width: number;
+    /**
+     * @deprecated use `width` only, avatars have a 1:1 aspect ratio
+     */
     height: number;
```
Comment (t3chguy) [inline src/components/views/avatars/BaseAvatar.tsx:43]: Can we just kill it? We're not a real SDK, confusion is worse than bigger diffs

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: src/components/views/avatars/BaseAvatar.tsx [hunk 7]
```diff
@@ -109,119 +112,32 @@ const BaseAvatar: React.FC<IProps> = (props) => {
...
-                }}
-                title={title}
-                alt=""
-                ref={inputRef}
-                data-testid="avatar-img"
-                {...otherProps}
-            />
-        );
-    }
+    return (
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/avatars/BaseAvatar.tsx:125]: The rule is disabled on this block

[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/avatars/BaseAvatar.tsx:125]: Unfortunately https://github.com/matrix-org/matrix-react-sdk/pull/11393/files#diff-f61169b723cd2eea456ca1c6fa8879aaf150f49060d77d4d8bf42257093e078fR114 applies to the whole block not just that one line

File: src/components/views/messages/RoomAvatarEvent.tsx [hunk 8]
```diff
@@ -82,7 +82,7 @@ export default class RoomAvatarEvent extends React.Component<IProps> {
                                 className="mx_RoomAvatarEvent_avatar"
                                 onClick={this.onAvatarClick}
                             >
-                                <RoomAvatar width={14} height={14} oobData={oobData} />
+                                <RoomAvatar width={16} oobData={oobData} />
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/messages/RoomAvatarEvent.tsx:85]: image

But it looks like it genuinely wants a 14px (OD) avatar?

File: src/components/views/avatars/BaseAvatar.tsx (hunk 6 above)
Comment (germain-gg) [reply src/components/views/avatars/BaseAvatar.tsx:43]: Done! I will rename `width` to `size` in a follow up PR

File: src/components/views/avatars/BaseAvatar.tsx (hunk 3 above)
Comment (germain-gg) [reply src/components/views/avatars/BaseAvatar.tsx:141]: Added to https://github.com/vector-im/compound-web/pull/57

Comment (germain-gg) [reply src/components/views/avatars/BaseAvatar.tsx:141]: Added to https://github.com/vector-im/compound-web/pull/57

**Pull request closed without merging**